src/Tests/dotnet-workload-update.Tests/GivenDotnetWorkloadUpdate.cs

```diff
@@ -142,20 +143,80 @@ public void GivenWorkloadUpdateItRollsBackOnFailedUpdate()
+        {
+            var mockWorkloadIds = new WorkloadId[] { new WorkloadId("xamarin-android") };
+            var cachePath = Path.Combine(_testAssetsManager.CreateTestDirectory(identifier: "cachePath").Path, "mockCachePath");
+            var parseResult = Parser.GetWorkloadsInstance.Parse(new string[] { "dotnet", "workload", "update", "--download-to-cache", cachePath });
+            (_, var command, var installer, _, var manifestUpdater, _) = GetTestInstallers(parseResult, installedWorkloads: mockWorkloadIds, includeInstalledPacks: true);
+
+            command.Execute();
+
+            // Manifest packages should have been 'downloaded' and used for pack resolution
+            manifestUpdater.DownloadManifestPackagesCallCount.Should().Be(1);
+            manifestUpdater.ExtractManifestPackagesToTempDirCallCount.Should().Be(1);
+            // 6 android pack packages need to be updated
+            installer.CachedPacks.Count.Should().Be(6);
+            installer.CachedPacks.Select(pack => pack.Id).Should().NotContain("Xamarin.Android.Sdk"); // This pack is up to date, doesn't need to be cached
+            installer.CachePath.Should().Be(cachePath);
+        }
+
+        [Fact]
+        public void GivenWorkloadUpdateItCanInstallFromOfflineCache()
+        {
+            var mockWorkloadIds = new WorkloadId[] { new WorkloadId("xamarin-android") };
+            var cachePath = "mockCachePath";
+            var parseResult = Parser.GetWorkloadsInstance.Parse(new string[] { "dotnet", "workload", "update", "--from-cache", cachePath });
+            (_, var command, var installer, _, _, var nugetDownloader) = GetTestInstallers(parseResult, installedWorkloads: mockWorkloadIds);
+
+            command.Execute();
+
+            installer.GarbageCollectionCalled.Should().BeTrue();
+            installer.CachePath.Should().Contain(cachePath);
+            installer.InstallationRecordRepository.WorkloadInstallRecord.Should().BeEquivalentTo(mockWorkloadIds);
+            installer.InstalledPacks.Count.Should().Be(8);
+            installer.InstalledPacks.Where(pack => pack.Id.Contains("Android")).Count().Should().Be(8);
+            nugetDownloader.DownloadCallParams.Count().Should().Be(0);
+        }
+
+        [Fact]
+        public void GivenWorkloadUpdateItPrintsDownloadUrls()
+        {
+            var mockWorkloadIds = new WorkloadId[] { new WorkloadId("xamarin-android") };
+            var parseResult = Parser.GetWorkloadsInstance.Parse(new string[] { "dotnet", "workload", "update", "--print-download-link-only" });
+            (_, var command, _, _, _, _) = GetTestInstallers(parseResult, installedWorkloads: mockWorkloadIds, includeInstalledPacks: true);
+
+            command.Execute();
+
+            _reporter.Lines.Should().Contain("==allPackageLinksJsonOutputStart==");
+            string.Join(" ", _reporter.Lines).Should().Contain("mock-url-xamarin.android.templates"); // New packs should be included
```

Comment:
nit: fluent assertion usually has a second parameter as reason. The reason will show up as part of the assertion failure

```c#
string.Join(" ", _reporter.Lines).Should().Contain("mock-url-xamarin.android.templates", "New` packs should be included");
```